instrumentation/opentelemetry-annotations-1.0/javaagent/src/test/groovy/WithSpanInstrumentationTest.groovy

```diff
@@ -370,4 +381,57 @@ class WithSpanInstrumentationTest extends AgentInstrumentationSpecification {
       }
     }
   }
+
+  def "instrument java6 class"() {
+    setup:
+    /*
+     class GeneratedJava6TestClass implements Runnable {
+       @WithSpan
+       public void run() {
+         TraceUtils.runUnderTrace("intercept", {})
+       }
+     }
+     */
+    Class<?> generatedClass = new ByteBuddy(ClassFileVersion.JAVA_V6)
+      .subclass(Object)
+      .name("GeneratedJava6TestClass")
+      .implement(Runnable)
+      .defineMethod("run", void.class, Modifier.PUBLIC).intercept(MethodDelegation.to(new Object() {
+        @RuntimeType
+        void intercept(@This Object o) {
+          TraceUtils.runUnderTrace("intercept", {})
+        }
+      }))
+      .visit(new MemberAttributeExtension.ForMethod()
+        .annotateMethod(AnnotationDescription.Builder.ofType(WithSpan).build())
+        .on(ElementMatchers.named("run")))
+      .make()
+      .load(getClass().getClassLoader())
+      .getLoaded()
```

Comment:
nice, I would have found a (random old) library that was compiled to Java 6 and tested instrumentation of that library, but this is better 👍